libtesting-ui/src/main/java/com/mapbox/navigation/testing/ui/MockLocationUpdatesRule.kt

```diff
@@ -0,0 +1,105 @@
+package com.mapbox.navigation.testing.ui
+
+import android.content.Context
+import android.location.Location
+import android.location.LocationManager
+import android.os.Build
+import android.os.SystemClock
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.platform.app.InstrumentationRegistry.getInstrumentation
+import org.junit.rules.TestWatcher
+import org.junit.runner.Description
+import java.util.Date
+
+private const val mockProviderName = LocationManager.GPS_PROVIDER
+
+/**
+ * Rule that sets up a mock location provider that can inject location samples
+ * straight to the device that the test is running on.
+ */
+class MockLocationUpdatesRule : TestWatcher() {
+
+    private val instrumentation = getInstrumentation()
+    private val appContext = (ApplicationProvider.getApplicationContext() as Context)
+
+    private val locationManager: LocationManager by lazy {
+        (appContext.getSystemService(Context.LOCATION_SERVICE) as LocationManager)
+    }
+
+    override fun starting(description: Description?) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            with(instrumentation.uiAutomation) {
+                val result = executeShellCommand(
+                    "appops set " +
+                        appContext.packageName +
+                        " android:mock_location allow"
+                )
+            }
+        } else {
+            throw RuntimeException("MockLocationUpdatesRule is supported on version codes >= Build.VERSION_CODES.M")
+        }
+
+        try {
+            locationManager.addTestProvider(
+                mockProviderName,
```

Comment:
nit: will add the parameter syntax here?

```
locationManager.addTestProvider(
    provider = mockProviderName,
    requiresNetwork = false, 
    requiresSatellite = false, 
    requiresCell = false, 
...
```

Reply:
`addTestProvider` is a Java API, it doesn't support named arguments.